### PR TITLE
H5 patches

### PR DESCRIPTION
**Pull request by David-OConnor**
This is a WIP. Currently, the changes in it, eg removing underscores on registers, is not working; open to ideas. Likely many more patches to apply after that part is fixed.

Edit: Per a discussion with Burbull, it seems the changes are taking on the GH "Memory map comparison" build, but not on my machine.

### CODE REVIEW

File: devices/common_patches/h5.yaml [hunk 1]
```diff
@@ -1 +1,78 @@
...
+  _delete:
+    - CNT
+  _modify:
+    CNT_alternate:
+      name: "CNT"
+
+TIM*:
+  _strip:
+    - "TIM*__"
+    - "TIM?*_"
```
Comment (burrbull) [inline devices/common_patches/h5.yaml:78]: I don't know what you wanted to do, but don't use `*` in `_strip` pattern.
`*` means any symbol<b>s</b>

```suggestion
TIM3:
  _delete:
    - CNT
  _modify:
    CNT_alternate:
      name: "CNT"

TIM*:
  _strip:
    - "TIM?_"
    - "TIM??_"
```

**Comment by David-OConnor** [comment]
Update: The build process has gotten the better of me; putting this on hold.

**Comment by David-OConnor** [comment]
This is in pretty good state now. Happy to merge, then patch up later A/R. Thanks Burbull for the stm32-rs and SVD2Rust patches!

**Review by adamgreig — APPROVED** [review]
thanks!

bors r+

**Pull request merged**